salt/minion.py

```diff
@@ -240,27 +240,39 @@ def resolve_dns(opts, fallback=True):
 
 
 def prep_ip_port(opts):
+    '''
+    parse host:port values from opts['master'] and return valid:
+        master: ip address or hostname as a string
+        master_port: (optional) master returner port as integer
+
+    e.g.:
+      - master: 'localhost:1234' -> {'master': 'localhost', 'master_port': 1234}
+      - master: '127.0.0.1:1234' -> {'master': '127.0.0.1', 'master_port' :1234}
+      - master: '[::1]:1234' -> {'master': '::1', 'master_port': 1234}
+      - master: 'fe80::a00:27ff:fedc:ba98' -> {'master': 'fe80::a00:27ff:fedc:ba98'}
+    '''
     ret = {}
     # Use given master IP if "ip_only" is set or if master_ip is an ipv6 address without
     # a port specified. The is_ipv6 check returns False if brackets are used in the IP
     # definition such as master: '[::1]:1234'.
     if opts['master_uri_format'] == 'ip_only' or salt.utils.network.is_ipv6(opts['master']):
-        ret['master'] = opts['master']
+        ret['master'] = str(ipaddress.ip_address(opts['master']))
```

Comment:
Just to note here that https://github.com/saltstack/salt/pull/49705 will add scope on IPv6, if any, once you `__str__` the object. That said, if  you have something like `fe80::1ff:fe23:4567:890a%eth2` in  as an input, you will get _exactly_ the same thing back.

And vice versa, without mentioned PR, if you have scope, now it will just crash (all python versions affected from 2.6 to 3.7).

Reply:
I submitted an upstream bug for ipaddress.IPv6Address() scoped link local addresses for a long term fix (or won't fix, if that's all we can get).

I think I want to make this PR depend on #49705 though.

Reply:
@aphor yeah, your should go after. I would expect some breakage, you should be able to resolve it though, as it is just imports in your case.